gcip/addons/container/jobs/crane.py

```diff
@@ -118,3 +118,65 @@ def push(
         job.prepend_scripts(*docker_client_config.get_shell_command())
 
     return job
+
+
+def pull(
+    *,
+    src_registry: Union[Registry, str],
+    image_name: str,
+    tar_path: Optional[str] = None,
+    image_tag: Optional[str] = None,
+    docker_client_config: Optional[DockerClientConfig] = None,
+    crane_image: Optional[Union[Image, str]] = None,
+) -> Job:
+    """
+    Creates a job to pull container image from remote container registry with `crane`.
+
+    Args:
+        src_registry (str): Registry URL to pull container image from.
+        image_name (str): Container image with namespace to pull from `src_registry`.
+            If `None` it defaults internally to `PredefinedVariables.CI_PROJECT_NAME`.
+        image_tag (str): Tag of the image which will be pulled.  If `None` it defaults internally to `PredefinedVariables.CI_COMMIT_TAG`
+            or `PredefinedVariables.CI_COMMIT_REF_SLUG` in order.
+        tar_path (Optional[str], optional): Path where to save the container image tarball.
+            If `None` it defaults internally to `PredefinedVariables.CI_PROJECT_DIR`. Defaults to None.
+        docker_client_config (Optional[DockerClientConfig], optional): Creates the Docker configuration file base on objects settings,
+            to authenticate against given registries. Defaults to a `DockerClientConfig` with login to the official Docker Hub
+            and expecting credentials given as environment variables `REGISTRY_USER` and `REGISTRY_LOGIN`.
+        crane_image (Optional[Union[Image, str]], optional): Container image which contains `crane` command.
+            Defaults to PredefindedImages.CRANE.
+
+    Returns:
+        Job: Returns a `gcip.Job`, with neccessary configuration to pull a container image from a remote registry and stores it in `tar_path`.
+            Job runs in ```stage=pull```
+    """
+    if not crane_image:
+        crane_image = PredefinedImages.CRANE
+
+    if not tar_path:
+        tar_path = PredefinedVariables.CI_PROJECT_DIR
+
+    image_path = image_name.replace("/", "_")
+
+    if not docker_client_config:
+        docker_client_config = DockerClientConfig()
+        docker_client_config.add_auth(registry=Registry.DOCKER)
+
+    if not image_tag:
```

Comment:
I should have mention this in the same comment as with `image_name` 😄 

It think it is a very special case, that the image tag of the image to pull is the same than as the tag/ref_slug of the target repository. It would be more obvious to force the user to set the tag of the source image or simply to omit the tag if not set.

Reply:
That is ok. I will make this mandatory too.

Reply:
Oh wait, i think it would be more natural, if the default would be `latest`. 
That is what all container tools are using if no tag is given. 
And if you want to pull an image maybe you want the `latest`.

Reply:
That is what I would first suggest. My second thought was to simply omit the tag (and let the underlying engine decide what to do)? That would be easier for us, because we do not overrule the users input.

What do you think what should happening, when the user does not set the parameter (by intention): Omitting the tag or defaulting him to latest?

Reply:
> What do you think what should happening, when the user does not set the parameter (by intention): Omitting the tag or defaulting him to latest?

Both would lead to `latest`.  I would suggest, defaulting to latest, because the implementation is just setting `image_tag: str = "latest"`. We do not have to change the underlying code.